cmd/sonarExecuteScan.go

```diff
@@ -84,6 +97,21 @@ func runSonar(config sonarExecuteScanOptions, client piperhttp.Downloader, runne
 		// handleArtifactVersion is reused from cmd/protecodeExecuteScan.go
 		sonar.addOption("sonar.projectVersion=" + handleArtifactVersion(config.ProjectVersion))
 	}
+	if len(config.ProjectKey) > 0 {
+		sonar.addOption("sonar.projectKey=" + config.ProjectKey)
+	}
+	if len(config.M2Path) > 0 && !SliceUtils.ContainsStringPart(config.Options, javaLibraries) {
+		sonar.addOption(javaLibraries + filepath.Join(config.M2Path, "**"))
+	}
+	if len(config.CoverageExclusions) > 0 && !SliceUtils.ContainsStringPart(config.Options, coverageExclusions) {
+		sonar.addOption(coverageExclusions + strings.Join(config.CoverageExclusions, ","))
+	}
+	if !SliceUtils.ContainsStringPart(config.Options, javaBinaries) {
+		addJavaBinaries()
+	}
+	if !SliceUtils.ContainsStringPart(config.Options, coverageReportPaths) {
```

Comment:
Most likely the report path is already defined in the `sonar-project.properties` file. 

How is the behavior on a multi module (not maven) project?

Reply:
The default behavior for the sonar scanner is to look for the reports at the default locations from the JaCoCo plugin for both the `report` and `report-aggregate` goal. This is a glob pattern as well. By default, the report location is "**/target/site/jacoco[-aggregate]/jacoco.xml".

Reply:
> Most likely the report path is already defined in the `sonar-project.properties` file.

Can you have another look? I hope to have provided a "good enough" solution for that.

Reply:
If no reports are found, the parameter will not be passed. So either a location is in sonar-project.properties where the step doesn't look, then nothing happens, or the same pattern is specified where the step finds the reports. Do you think this is OK?